### PR TITLE
feat: integration test for swap

### PR DESCRIPTION
**Pull request by zarboq**
# Pull Request type
- Feature
- Testing

## What is the current behavior?

There is no integration test for swap.

## What is the new behavior?

Integration test for swap & fixing swap workflow mainly, still have to fix oracle account signature check. Merging bc lot of fixes have been made that will be needful for other tests. Will continue in another PR.

### CODE REVIEW

File: src/oracle/oracle_utils.cairo [hunk 1]
```diff
@@ -117,6 +117,7 @@ fn COMPACTED_PRICE_INDEX_BITMASK() -> u128 {
     BITMASK_8
 }
 
+use debug::PrintTrait;
```
Comment (Sk8erboi84) [inline src/oracle/oracle_utils.cairo:120]: add that in the top of the file

File: src/exchange/order_handler.cairo [hunk 2]
```diff
@@ -108,6 +107,7 @@ mod OrderHandler {
     use starknet::ContractAddress;
     use starknet::{get_caller_address, get_contract_address};
     use array::ArrayTrait;
+    use debug::PrintTrait;
```
Comment (Sk8erboi84) [inline src/exchange/order_handler.cairo:110]: No need

**Review by Sk8erboi84 — CHANGES_REQUESTED** [review]

File: src/market/market_utils.cairo [hunk 3]
```diff
@@ -3,7 +3,7 @@
 // *************************************************************************
 // Core lib imports.
 use starknet::{ContractAddress, get_caller_address, get_block_timestamp, contract_address_const};
-
+use debug::PrintTrait;
```
Comment (Sk8erboi84) [inline src/market/market_utils.cairo:6]: Same here, you can remove the PrintTrait

**Review by Sk8erboi84 — CHANGES_REQUESTED** [review]

File: src/oracle/oracle.cairo [hunk 4]
```diff
@@ -678,12 +686,12 @@ mod Oracle {
                             report_info.min_price, report_info.max_price
                         );
                     }
-                    oracle_utils::validate_signer(
-                        self.get_salt(),
-                        report_info,
-                        *signatures_span.at(inner_cache.signature_index),
-                        signers_span.at(j)
-                    );
+                    // oracle_utils::validate_signer(
```
Comment (Sk8erboi84) [inline src/oracle/oracle.cairo:689]: Add a TODO if you are gonna modify it later

File: src/pricing/error.cairo [hunk 5]
```diff
@@ -13,5 +13,11 @@ mod PricingError {
         data.append(short_open_interest.into());
         panic(data)
     }
-    const USD_DELTA_EXCEEDS_POOL_VALUE: felt252 = 'usd_delta_exceeds_pool_value';
+
+    fn USD_DELTA_EXCEEDS_POOL_VALUE(usd_delta: felt252, pool_usd_for_token: u128) {
+        let mut data = array!['usd_delta_exceeds_pool_value'];
+        // data.append(usd_delta.into());
```
Comment (Sk8erboi84) [inline src/pricing/error.cairo:19]: Why comment that?

**Review by Sk8erboi84 — CHANGES_REQUESTED** [review]

File: src/pricing/error.cairo (hunk 5 above)
Comment (zarboq) [reply src/pricing/error.cairo:19]: It's crashing, couldn't find why

Comment (Sk8erboi84) [reply src/pricing/error.cairo:19]: If you think we are gonna reuse it later add a TODO crashing so we can fix it later and not forget

**Review by Sk8erboi84 — APPROVED** [review]

**Pull request merged**